Factory/Factory.php

```diff
@@ -51,4 +51,13 @@ public function getRegistry()
     {
         return $this->registry;
     }
+
+    /**
+     * @param $builderName
+     * @return bool true if this factory contains a builder with the given name
+     */
+    public function hasBuilder($builderName)
```

Comment:
this can be achieved with `getRegistry()->has()` do you think this method is still worth adding?

Reply:
Worth it. It helps keep things from breaking the Law of Demeter

Reply:
BS"D

yeah the thought makes sense. The thing is since the registry is available through a public method, adding a convenience method that does some of the calls for you does not affect the law of demeter since its already accessible. Its the same reason why security through obscurity is not real security.